src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -319,6 +310,20 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         }
 
         let place = self.codegen_place(&mut bx, location);
+        if let ty::InstanceDef::Intrinsic(_) = drop_fn.def {
+            let callee_ty = drop_fn.ty(bx.tcx());
+            let fn_ty = bx.fn_type_of_instance(&drop_fn);
+            let dest = bx.const_undef(bx.type_ptr_to(bx.memory_ty(&fn_ty.ret)));
+            let place_operand =
+                OperandRef::from_immediate_or_packed_pair(&mut bx, place.llval,
```

Comment:
I'm not sure this is fully correct, but it passes all the tests. I'm also not sure what to do with `place.llextra`.